lib/EarlyFunctionAnalysis/Outliner.cpp

```diff
@@ -425,9 +424,12 @@ void Outliner::createAnyPCHooks(OutlinedFunction *OutlinedFunction) {
   }
 }
 
-OutlinedFunction Outliner::outline(llvm::BasicBlock *Entry) {
+OutlinedFunction
+Outliner::outline(llvm::BasicBlock *Entry, CallHandler *Handler) {
   using namespace llvm;
 
+  TheCallHandler = Handler;
```

Comment:
This is sad. I'm starting to think that just hoisting CEAC is much more elegant.

Otherwise, the proper way to do this would be:
1. drop the `TheCalleHandler` field;
2. create a `OutlinerImpl` `class` that has as field `TheCallHandler`;
3. Have `Outliner::outline` create `OutlinerImpl` and invoke `run`;
4. Move all the methods using `TheCallHandler` into `OutlinerImpl`.

Reply:
i just slapped it as a argument everywhere it was needed